onset_maproom/tests/test_calc.py

```diff
@@ -4,24 +4,2025 @@
+        4.226911,
+        1.177538,
+        0.964985,
+        1.286437,
+        2.70218,
+        0.792248,
+        1.2045,
+        0.446366,
+        0.516156,
+        0.357902,
+        0.389365,
+        0.382511,
+        1.626637,
+        0.545314,
+        0.0,
+        0.9319,
+        0.0,
+        0.509477,
+        0.120703,
+        0.128327,
+        2.925095,
+        1.308689,
+        0.34651,
+        4.402626,
+        0.482099,
+        0.613916,
+        1.956849,
+        0.0,
+        0.299069,
+        0.169879,
+        0.095796,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.130293,
+        0.0,
+        0.20588,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.154896,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.255222,
+        0.126956,
+        0.10043,
+        0.006866,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.069079,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.001652,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.211269,
+        5.749138,
+        5.437428,
+        3.361074,
+        0.001031,
+        0.085564,
+        0.078808,
+        2.277596,
+        0.114164,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        4.02564,
+        1.322553,
+        0.0,
+        0.2692,
+        0.0,
+        0.0,
+        1.737253,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.0,
+        0.451228,
+        0.054314,
+        0.333831,
+        0.637705,
+        6.406705,
+    ]
+    precip = xr.DataArray(values, dims=["T"], coords={"T": t}).rename("precip")
+
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+    assert dts["group"].size == 5
+    assert (
+        dts.seasons_ends
+        == xr.DataArray(
+            [
+                "2001-02-28T00:00:00.000000000",
```

Comment:
One thing that's very important when writing tests: always perturb the test and see it fail, so you know that you're actually testing what you think you're testing. I tried changing `2001-02-28` in this line to `2001-02-27`, and the test still passes.

In this case, the test passes because you're asserting `ds.all` instead of `ds.all()`. `ds.all` is a function, which is a "truthy" value, so the assertion passes no matter what is in `ds`. You did the same further down. I didn't notice this when reading the code, I only found it after perturbing the setup and seeing that the test still passed.